File: Programming Basics with Python/05 While loop/More exercises/02_Report_System.py
money_target = int(input())
money_collect = 0
transaction_counter = 0
cash = 0
card = 0
transaction_cash = 0
transaction_card = 0
while money_target > money_collect:
    transaction_input = input()
    if transaction_input == "End":
        print("Failed to collect required money for charity.")
        break
    transaction_counter += 1
    if (transaction_counter % 2 == 1 and int(transaction_input) > 100) or (
            transaction_counter % 2 == 0 and int(transaction_input) < 10):
        print("Error in transaction!")
    elif transaction_counter % 2 == 1 and int(transaction_input) <= 100:
        print("Product sold!")
        transaction_cash += 1
        cash += int(transaction_input)
        money_collect += int(transaction_input)
    elif transaction_counter % 2 == 0 and int(transaction_input) >= 10:
        print("Product sold!")
        transaction_card += 1
        card += int(transaction_input)
        money_collect += int(transaction_input)
if money_target <= money_collect:
    print(f"Average CS: {(cash / transaction_cash):.2f}\nAverage CC: {(card / transaction_card):.2f}")

File: JS Basics/05 While-Loop - More Exercises/02. Report System.js
function report(input) {
    let targetSum = Number(input[0]);
    let index = 1;
    let cardPay = 0;
    let cardCounter = 0;
    let cashPay = 0;
    let cashCounter = 0;
    let transactionCounter = 0;
    let moneyCollect = 0;
    while (true) {
        let product = input[index];
        transactionCounter++;
        if (product == "End") {
            console.log("Failed to collect required money for charity.");
            break;
        }
        product = Number(product);
        if (transactionCounter % 2 == 0) {
            if (product >= 10) {
                cardPay += product;
                cardCounter++;
                moneyCollect += product;
                console.log("Product sold!");
            }
            else {
                console.log("Error in transaction!");
            }
        }
        else {
            if (product <= 100) {
                cashPay += product;
                cashCounter++;
                moneyCollect += product;
                console.log("Product sold!");
            }
            else {
                console.log("Error in transaction!");
            }
        }
        index++;
        if (moneyCollect >= targetSum) {
            let averageCS = cashPay / cashCounter;
            let averageCC = cardPay / cardCounter;
            console.log(`Average CS: ${averageCS.toFixed(2)}`);
            console.log(`Average CC: ${averageCC.toFixed(2)}`);
            break;
        }
    }
}
report([500, 120, 8, 63, 256, 78, 317])